examples-gallery/plot_parameter_estimation.py

```diff
@@ -0,0 +1,227 @@
+
+kd = sm.Matrix([u1 - x1.diff(), u2 - x2.diff(), u3 - x3.diff(), u4 - x4.diff()])
+
+q_ind = [x1, x2, x3, x4]
+u_ind = [u1, u2, u3, u4]
+
+KM = me.KanesMethod(N, q_ind, u_ind, kd_eqs=kd)
+fr, frstar = KM.kanes_equations(bodies, forces)
+eom = kd.col_join(fr + frstar)
+sm.pprint(eom)
+
+rhs = KM.rhs()
+
+qL = q_ind + u_ind
+pL = [m, c, k, l0]
+
+rhs_lam = sm.lambdify(qL + pL, rhs)
+
+
+def gradient(t, x, args):
+    return rhs_lam(*x, *args).reshape(8)
+
+t0, tf = 0, 20
+num_nodes = 500
+times = np.linspace(t0, tf, num_nodes)
+t_span = (t0, tf)
+
+x0 = np.array([2, 3, 4, 5, 0, 0, 0, 0])
+pL_vals = [1.0, 0.25, 1.0, 1.0]
+
+resultat1 = solve_ivp(gradient, t_span, x0, t_eval = times, args=(pL_vals,))
+resultat = resultat1.y.T
+print('Shape of result: ', resultat.shape)
+print(' the message is: ', resultat1.message)
+
+noisy = []
+np.random.seed(123)
+for i in range(4):
+    noisy.append(resultat[:, i] + np.random.randn(resultat.shape[0]) * 0.5 +
+        np.random.randn(1))
+
+fig, ax = plt.subplots(figsize=(10, 5))
+for i in range(4):
+    ax.plot(times, resultat[:, i], label=f'x{i+1}')
+    ax.plot(times, noisy[i], label=f'noisy x{i+1}', lw=0.5)
+plt.xlabel('Time')
+ax.set_title('Noisy measurements of the position of the mass')
+ax.legend();
+
+plt.show()
+
+
+# %%
+# Set up the Estimation Problem.
+# --------------------------------
+#
+# If some measurement is considered more reliable, its weight can be increased.
+#
+# objective = :math:`\int_{t_0}^{t_f} (weight_1 (x_1 - noisy_{x_1})^2 + weight_2 (x_2 - noisy_{x_2})^2 + weight_3 (x_3 - noisy_{x_3}))^2 + weight_4 (x_4 - noisy_{x_4})^2)\, dt`
+#
+state_symbols = [x1, x2, x3, x4, u1, u2, u3, u4]
+unknown_parameters = [c, k]
+
+interval_value = (tf - t0) / (num_nodes - 1)
+par_map = {m: pL_vals[0], l0: pL_vals[3]}
+
+weight =[1, 1, 1, 1]
+def obj(free):
+    return interval_value *np.sum((weight[0] * free[:num_nodes] - noisy[0])**2 +
+            weight[1] * (free[num_nodes:2*num_nodes] - noisy[1])**2 +
+            weight[2] * (free[2*num_nodes:3*num_nodes] - noisy[2])**2 +
+            weight[3] * (free[3*num_nodes:4*num_nodes] - noisy[3])**2
+)
+
+
+def obj_grad(free):
+    grad = np.zeros_like(free)
+    grad[:num_nodes] = 2 * weight[0] * interval_value * (free[:num_nodes] - noisy[0])
+    grad[num_nodes:2*num_nodes] = 2 * weight[1] * (interval_value *
+                    (free[num_nodes:2*num_nodes] - noisy[1]))
+    grad[2*num_nodes:3*num_nodes] = 2 * weight[2] * (interval_value *
+                    (free[2*num_nodes:3*num_nodes] - noisy[2]))
+    grad[3*num_nodes:4*num_nodes] = 2 * weight[3] * (interval_value *
+                    (free[3*num_nodes:4*num_nodes] - noisy[3]))
+    return grad
+
+instance_constraints = (
+    x1.subs({t: t0}) - x0[0],
+    x2.subs({t: t0}) - x0[1],
+    x3.subs({t: t0}) - x0[2],
+    x4.subs({t: t0}) - x0[3],
+    u1.subs({t: t0}) - x0[4],
+    u2.subs({t: t0}) - x0[5],
+    u3.subs({t: t0}) - x0[6],
+    u4.subs({t: t0}) - x0[7],
```

Comment:
You generally don't want instance constraints in parameter estimation problems. This overly constrains the solution.

Reply:
> You generally don't want instance constraints in parameter estimation problems. This overly constrains the solution.

Sorry, I missed this in the PR I just started. Will correct it.

Reply:
My bet is that if you remove these, you'll get better parameter estimates.

Reply:
>  Question: Would you not know the initial conditionsd of the experiment, like im my case you would know how much you extended the mas, and at what initial speed it started?

You know the initial conditions only as good as your error filled measurement might say. If you force the initial condition to the value of a measurement that has error, then you are not allowing for solutions that minimize the overall error.

Reply:
Makes sense!
When I give reasonable bounds it finds good values.
I will try to incorporate all you suggested tomorrow.

Reply:
Question: does it make sense to estimate continuous parameters this way? What I mean: say, I apply a time varying force to this sytem and try to estimate what it looks like?

Reply:
If I understand you, I think you can.